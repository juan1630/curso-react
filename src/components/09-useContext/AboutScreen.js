import React, { useContext } from 'react'
import { UserContex } from './UserContext'

export const AboutScreen = () => {

  const { user, setUser } =  useContext( UserContex );

  const handleClick = () => {
      // el context hace el redibujo de los componentes
    setUser({});
  }

    return (
        <div>
             <h1> about screen</h1>
             { user.name }
             <hr/>
             <pre>
                 { JSON.stringify(user, null, 3)}
             </pre>


             <button  className="btn btn-warning btn-block" 
                onClick={ handleClick }
             >
                    Log out
             </button>
        </div>
    )
}
