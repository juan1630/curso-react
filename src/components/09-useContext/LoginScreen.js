import React, { useContext } from 'react';
import { UserContex } from './UserContext';


export const LoginScreen = () => {
    // leer la referencia del context
    const {  setUser } = useContext( UserContex );


    //extraer el setUser
    // llamar la funcion setUser y settear la data el user 
    const user = {
        id: 123,
        name:'Juan'
    }
    // id y name
    return (
        <div>
           <h1> Login screen</h1>
           
           <button 
                className="btn btn-info btn-block"
                onClick={ () => setUser ( user  )  }
                >
                Enviar 
           </button>
        </div>
    )
}
