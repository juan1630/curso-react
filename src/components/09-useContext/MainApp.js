import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UserContex } from './UserContext'

export const MainApp = () => {

    
   const [user, setUser] = useState({});

    return (
         
        // Lo que enmarca a los componentes se puede utilizar en los demas componentes
            <UserContex.Provider value={{
                user,
                setUser
            }}  >
                <AppRouter/>

            </UserContex.Provider>
        
    )
}
