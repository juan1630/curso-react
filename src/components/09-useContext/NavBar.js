import React from 'react';
import { Link } from 'react-router-dom';

export const NavBar = () => {
    return (


        <nav className="navbar navbar-expand-lg navbar-light bg-light" >
            <div className="container-fluid" >
            <ul className="navbar-nav" >
                
                <li className="nav-item" >
                    <Link  className="nav-link" to="./" >
                    
                        Home
                    </Link>
                </li>
                <li className="nav-item"  >
                    <Link className="nav-link" to="/about"  >
                        About
                    
                    </Link>
                </li>
                <li className="nav-item" >
                    <Link className="nav-link" to="/login" >
                        Login
                    
                    </Link>
                    
                </li>
            </ul>
            </div>
        </nav>
    )
}
