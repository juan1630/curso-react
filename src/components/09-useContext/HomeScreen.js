import React from 'react'
import { NavBar } from './NavBar'

export const HomeScreen = () => {
    return (
        <div>

            <h1> Home screen</h1>
        </div>
    )
}
