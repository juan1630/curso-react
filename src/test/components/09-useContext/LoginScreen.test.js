import { shallow } from 'enzyme';
import React from 'react';
import {LoginScreen} from '../../../components/09-useContext/LoginScreen';
import { UserContex } from '../../../components/09-useContext/UserContext';

describe('Pruebas en el componente LoginScreen', () => {
    

    const setUser = jest.fn()

    const wrapper = shallow(
        <UserContex.Provider  value={{
            setUser
        }} >

            <LoginScreen/>
        </UserContex.Provider>
    )
    
    test('Debe de mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe de ejecutar el setUser con el argumento esperado', ()=> {
       
        wrapper.find('button').prop('onClick')();

        expect( wrapper ).toHaveBeenNthCalledWith({
            
                id: 123,
                name:'Juan'
                    })
    });
    
})
