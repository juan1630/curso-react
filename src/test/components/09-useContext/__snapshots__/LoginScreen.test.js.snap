// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente LoginScreen Debe de mostrarse correctamente 1`] = `<LoginScreen />`;
